orb-ui/src/sound/mod.rs

```diff
@@ -408,13 +496,59 @@ impl Player for Fake {
             tracing::error!("Sound file not found: {:?}", sound_type);
         }
     }
+
+    /// Finds the sound file path for the given sound type and sends it to the
+    /// sound player **only if the sink is empty**
+    fn try_queue(&self, sound_type: Type) -> Result<bool> {
+        if let Ok(sink) = self.sink.lock() {
+            if sink.empty() {
+                if let Some(sound_file) = self.sound_files.get(&sound_type) {
+                    if let Some(sound_file) = sound_file.value() {
+                        if let Ok(mut tx_queue) = self.queue_file.lock() {
+                            tx_queue
+                                .try_send(sound_file.clone())
+                                .wrap_err("Failed to queue sound")?;
+                            return Ok(true);
+                        }
+                    } else {
+                        tracing::error!(
+                            "Sound file {:?} doesn't have a known file path",
+                            sound_type
+                        );
+                    }
+                } else {
+                    tracing::error!("Sound file not found: {:?}", sound_type);
+                }
+            }
+        }
+        Ok(false)
+    }
+
+    fn set_volume(&self, volume_percent: u64) {
+        if let Ok(sink) = self.sink.lock() {
+            sink.set_volume((volume_percent as f64 / 100_f64) as f32);
+        } else {
+            tracing::error!("Failed to lock sink to set sound volume");
```

Comment:
This failure mode will go away when we get rid of the mutex, but for future reference:

If the mutex fails to lock, it means it is poisoned and something very wrong happened. Logging is probably not the right behavior, I would panic.